ui/src/components/PollVoteMain/PollVoteMain.jsx

```diff
@@ -0,0 +1,54 @@
+import React, { useState } from 'react';
+import VoteCompletePopup from '../VoteCompletePopup/VoteCompletePopup';
+import '../PollVoteHome/PollVoteHome.css';
+import './PollVoteMain.css';
+
+const PollVoteMain = (props) => {
+    const [selection, setSelection] = useState(-1);
+    const [popupVisible, setPopupVisible] = useState(false);
+    const { pollTitle, pollDescription, pollOptions, deadline, setCurrPage } = props;
+
+    /* Check the deadline again, and check whether the user selected an option */
+    const submitVote = () => {
+        const currDate = new Date();
+        const deadlineDate = new Date(deadline);
+        if (currDate > deadlineDate) {
+            alert("The deadline to vote has passed. Click ok to be redirected to the vote page.");
+            setCurrPage(0);
+        }
+        else if (selection == -1) { alert("Please select an option"); }
+        else {
+            document.getElementById('vote-main-body').style.filter = 'blur(10px)';
+            setPopupVisible(true);
+        }
+    }
+
+    let optionInd = 0;
+    const optionsSection = pollOptions ? pollOptions.map((option) => {
+        const optionId = "option" + optionInd;
+        optionInd++;
+        return (
+            <div className="vote-option-box">
+                <label for={optionId}>{option.optionName}</label>
+                <input type="radio" name="pollOption" id={optionId} onClick={() => setSelection(optionInd - 1)} />
+            </div>
+        );
+    }) : <div></div>
```

Comment:
The `optionInd` variable is your index for the pollOptions array. Fortunately, you don't need it, since the `map` function already provides one for you!
```suggestion
    const optionsSection = pollOptions ? pollOptions.map((option, optionInd) => {
        const optionId = "option" + optionInd;
        return (
            <div className="vote-option-box">
                <label for={optionId}>{option.optionName}</label>
                <input type="radio" name="pollOption" id={optionId} onClick={() => setSelection(optionInd)} />
            </div>
        );
    }) : <div></div>
```
There are many little niceties like these through JavaScript's functions, and they're easily found on [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map). Neat, right?

P.S: I changed `setSelection` to use `optionInd` in the suggestion because leaving the -1 causes the selection thing not to work. I assume selections are zero-indexed, right?